src/promql/src/extension_plan/instant_manipulate.rs

```diff
@@ -0,0 +1,720 @@
+        }
+    }
+
+    pub fn to_execution_plan(&self, exec_input: Arc<dyn ExecutionPlan>) -> Arc<dyn ExecutionPlan> {
+        Arc::new(InstantManipulateExec {
+            start: self.start,
+            end: self.end,
+            lookback_delta: self.lookback_delta,
+            interval: self.interval,
+            time_index_column: self.time_index_column.clone(),
+            input: exec_input,
+            metric: ExecutionPlanMetricsSet::new(),
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct InstantManipulateExec {
+    start: Millisecond,
+    end: Millisecond,
+    lookback_delta: Millisecond,
+    interval: Millisecond,
+    time_index_column: String,
+
+    input: Arc<dyn ExecutionPlan>,
+    metric: ExecutionPlanMetricsSet,
+}
+
+impl ExecutionPlan for InstantManipulateExec {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.input.schema()
+    }
+
+    fn output_partitioning(&self) -> Partitioning {
+        self.input.output_partitioning()
+    }
+
+    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
+        self.input.output_ordering()
+    }
+
+    fn maintains_input_order(&self) -> bool {
+        true
+    }
+
+    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
+        vec![self.input.clone()]
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        children: Vec<Arc<dyn ExecutionPlan>>,
+    ) -> DataFusionResult<Arc<dyn ExecutionPlan>> {
+        assert!(!children.is_empty());
+        Ok(Arc::new(Self {
+            start: self.start,
+            end: self.end,
+            lookback_delta: self.lookback_delta,
+            interval: self.interval,
+            time_index_column: self.time_index_column.clone(),
+            input: children[0].clone(),
+            metric: self.metric.clone(),
+        }))
+    }
+
+    fn execute(
+        &self,
+        partition: usize,
+        context: Arc<TaskContext>,
+    ) -> DataFusionResult<SendableRecordBatchStream> {
+        let baseline_metric = BaselineMetrics::new(&self.metric, partition);
+
+        let input = self.input.execute(partition, context)?;
+        let schema = input.schema();
+        let time_index = schema
+            .column_with_name(&self.time_index_column)
+            .expect("time index column not found")
+            .0;
+        Ok(Box::pin(InstantManipulateStream {
+            start: self.start,
+            end: self.end,
+            lookback_delta: self.lookback_delta,
+            interval: self.interval,
+            time_index,
+            schema,
+            input,
+            metric: baseline_metric,
+        }))
+    }
+
+    fn fmt_as(&self, t: DisplayFormatType, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match t {
+            DisplayFormatType::Default => {
+                write!(
+                    f,
+                    "PromInstantManipulatorExec: range=[{}..{}], lookback=[{}], interval=[{}], time index=[{}]",
```

Comment:
Forgot to modify it? `PromInstantManipulatorExec` -> `InstantManipulateExec`.

Reply:
Yes 🥲...